### PR TITLE
[QUIZ-674]: Configure Results Summary Text Dashboard Experience

### PR DESCRIPTION
**Pull request by Gull-Dev**
Set up the '@mention' tag in the admin panel to represent all selected options from the quiz. This feature allows for a more personalized summary based on users choices

Ref: 
1) https://linear.app/constructor/issue/QUIZ-661/result-page-configure-results-summary-text-dashboard-experience
2) https://linear.app/constructor/issue/QUIZ-657/result-page-configure-results-summary-text-dashboard-experience-basic

### CODE REVIEW

File: src/stories/Quiz/Hooks/index.tsx [hunk 1]
```diff
@@ -33,7 +35,20 @@ export default function HooksTemplate(args) {
       ? state.quiz.resultsConfig?.desktop?.title?.text || 'Here are your results'
       : '';
     const resultsDescription = state.quiz.resultsConfig?.desktop?.description?.text || '';
-    const matchedOptions = state.quiz.matchedOptions || [];
+    const matchedOptions = state.quiz.matchedOptions || '';
```
[the remark this answers is not in the document]
Comment (Gull-Dev) [reply src/stories/Quiz/Hooks/index.tsx]: [443914e](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/168/commits/443914ed95134bcdbdae664eb2e2ed074368e8ee)

File: src/utils.tsx [hunk 2]
```diff
@@ -188,3 +188,11 @@ export function convertPrimaryColorsToString(primaryColorStyles: PrimaryColorSty
     --primary-color-l: ${primaryColorStyles['--primary-color-l']}; 
   }`;
 }
+
+export function formatMatchedOptions(options, separator, lastSeparator) {
```
[the remark this answers is not in the document]
Comment (Gull-Dev) [reply src/utils.tsx]: [443914e](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/168/commits/443914ed95134bcdbdae664eb2e2ed074368e8ee)

**Review by vladconstructor — APPROVED** [review]
Good from my side!
Would love to hear from others as well

File: src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx [hunk 3]
```diff
@@ -15,14 +17,28 @@ function ResultFiltersAndShare({
 }: ResultFiltersAndShareProps) {
   const { state } = useContext(QuizContext);
   const matchedOptions = state?.quiz?.matchedOptions || [];
+  const {
```
Comment (mocca102) [inline src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx:20]: I see the logic here is duplicated again in HooksTemplate. Can we abstract it in a function or a utility hook instead?

File: src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx [hunk 4]
```diff
@@ -15,14 +17,28 @@ function ResultFiltersAndShare({
...
+    text = '',
+    is_active: isActive,
+    items_separator: itemsSeparator,
+    last_separator: lastSeparator,
+  } = state?.quiz?.resultsConfig?.desktop?.response_summary || {};
```
Comment (mocca102) [inline src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx:25]: getting a type error here `Property 'response_summary' does not exist on type 'ViewportResultsConfig' `

File: src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx (hunk 3 above)
Comment (mocca102) [reply src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx:20]: What do you think about making summary it's own text component? 🤔

Comment (Gull-Dev) [reply src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx]: Yes, using a Component does seem to be a more effective approach. I have completed the refactoring using the Component for the summary [4408a10](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/168/commits/4408a10b28051fee058364be910c0063b4d9d002)

File: src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx (hunk 4 above)
Comment (Gull-Dev) [reply src/components/ResultFiltersAndShare/ResultFiltersAndShare.tsx]: you should execute `npm i `, as there are recent changes from internal libraries related to types

File: src/stories/Quiz/Component/QuizResults/QuizResultsDecorator.tsx [hunk 5]
```diff
@@ -37,3 +37,17 @@ export function QuizResultsPrimaryDecorator(Story: any) {
...
 }
+
+export function QuizResultsWithSummaryDecorator(Story: any) {
+  const contextValue = useMockContextValue(undefined, { withSummary: true });
+
+  return (
+    <div className='cio-quiz'>
+      <QuizContext.Provider value={contextValue}>
+        <div>
+          <Story Component={Story} />
```
Comment (mocca102) [inline src/stories/Quiz/Component/QuizResults/QuizResultsDecorator.tsx:48]: <StoryPreview /> Instead of stories for all window sizes

**Review by mocca102 — APPROVED** [review]
LGTM!

File: src/stories/Quiz/Component/QuizResults/QuizResultsDecorator.tsx (hunk 5 above)
Comment (Gull-Dev) [reply src/stories/Quiz/Component/QuizResults/QuizResultsDecorator.tsx]: [cbb7699](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/168/commits/cbb7699d030b8b5ea880dfc70c4b74c58716ceec)

**Pull request merged**